### PR TITLE
Obtain Player Tokens

### PR DESCRIPTION
**Pull request by grace000**
https://trello.com/c/bEBgCxUZ/2-ability-to-get-token-for-both-players

### CODE REVIEW

File: lib/run.rb [hunk 1]
```diff
@@ -16,3 +17,11 @@ def prompt_token_selection
 def get_player_token
     input = Input.new.get_input
 end
+
+def display_board
+    board = BoardPresenter.new.display_board
```
Comment (BWoodfork) [inline lib/run.rb]: Another unnecessary variable assignment

File: lib/run.rb [hunk 2]
```diff
@@ -16,3 +17,11 @@ def prompt_token_selection
...
+end
+
+def assign_token
+    get_player_token == "X" ? "O": "X"
+end
```
Comment (BWoodfork) [inline lib/run.rb]: What's this function being used for right now? / Where is it being called?

File: lib/run.rb (hunk 1 above)
Comment (grace000) [reply lib/run.rb]: Removed the assignment

File: lib/run.rb (hunk 2 above)
Comment (grace000) [reply lib/run.rb]: Not currently being called anywhere, but I need this assignment to start the game.

File: spec/board_presenter_spec.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+describe BoardPresenter do
+    before(:each) do
+        @board_presenter = BoardPresenter.new
+    end
+
+    describe "#display_board" do
+        it "renders empty board if there are no player marks" do
+            board = Board.new
+
+            board.moves.length == 0
```
Comment (BWoodfork) [inline spec/board_presenter_spec.rb]: `==` is an equality check. Not a variable assignment. What's the purpose of this line?

File: lib/run.rb [hunk 4]
```diff
@@ -1,18 +1,28 @@
...
 end
+
+def display_board
+    BoardPresenter.new.display_board
+end
+
+def assign_token
+    get_player_token == "X" ? "O": "X"
+end
+
```
Comment (BWoodfork) [inline lib/run.rb]: So, when you run `ruby run.rb` .. What currently happens? I don't see anything in here where these methods are actually called. So the acceptance criteria isn't actually met. Am I missing something?

File: spec/board_presenter_spec.rb (hunk 3 above)
Comment (grace000) [reply spec/board_presenter_spec.rb]: While the 'moves' array is empty, the board should also be empty. So future tests would have board moves, and the board should correlate with the moves that have been made.

Comment (BWoodfork) [reply spec/board_presenter_spec.rb]: Ok. What's the purpose of this line in the test though? It's not assigning anything or asserting anything.

Comment (BWoodfork) [reply spec/board_presenter_spec.rb]: To better put things into perspective, if you delete this line. Your tests will still pass.

Comment (grace000) [reply spec/board_presenter_spec.rb]: Absolutely! That was for my own sanity check. I have deleted that line.

File: lib/input.rb [hunk 5]
```diff
@@ -1,5 +1,10 @@
 class Input
-    def get_input
-        $stdin.gets.chomp.upcase
+    def get_num_players
+        $stdin.gets.chomp
     end
+
+    def get_player_token
+        $stdin.gets.chomp
+    end
```
Comment (BWoodfork) [inline lib/input.rb]: Both of these functions do the same thing.

Comment (grace000) [reply lib/input.rb]: I will delete this.

File: lib/tic_tac_toe.rb [hunk 6]
```diff
@@ -0,0 +1,21 @@
+require_relative './command_line_in'
+require_relative './prompt'
+require_relative './board_presenter'
+
+class TicTacToe
+    def assign_token(taken_token)
+        taken_token == 'X' ? "O" : "X"
```
Comment (BWoodfork) [inline lib/tic_tac_toe.rb]: Another variable that does nothing

Comment (grace000) [reply lib/tic_tac_toe.rb]: Yes, the token should be abstracted and stored elsewhere. I'll delete this line.

**Pull request merged**